### ISSUE TITLE
syphon plugin interaction with certain skydomes blacks out the skytexture

### ISSUE THREAD

**Issue by VSLABKRS** [Syphon/Unity3D#23]
![screen shot 2016-05-03 at 20 40 26](https://cloud.githubusercontent.com/assets/19174463/14994299/679c8810-116f-11e6-8335-9b248455e7f2.png)
![screen shot 2016-05-03 at 20 40 47](https://cloud.githubusercontent.com/assets/19174463/14994300/680e2416-116f-11e6-859a-9477db104fc5.png)

the images speak for themselves. as soon as SyphonServerTextureCustomResolution script is activated the textured skydome gets lost and is blacked out in the syphon output. does anybody know how to fix that?

**Comment by vade** [comment]
This could be a premultiplied vs un premultiplied alpha issue? I'd experiment with that.

**Comment by VSLABKRS** [comment]
where do i set these values?

**Comment by vade** [comment]
This has to do with the blending in Unity and the construction of the image you send to it via Syphon. I have no idea, you're in control :)

**Comment by VSLABKRS** [comment]
So I checked all blend-modes in the skyshaders without having luck. I then looked
at the Syphon Server Custom Resolution script and changed from ARGB32 to RGB565 in this line:

customRenderTexture = new RenderTexture(renderWidth, renderHeight, 0, RenderTextureFormat.RGB565);

Now the alphachannels are fine, but the color resolution is bad. Do you have a workaround for that?

![screen shot 2016-05-04 at 19 55 07](https://cloud.githubusercontent.com/assets/19174463/15023917/d7092db6-1232-11e6-81fc-a617fb7616f7.png)

**Comment by vade** [comment]
Thats likely not going to fix it.

Your issue is either your client is expecting color in the form of premultiplied:

vec4 RGB = vec4(color.r * color.a, color.b * color.a, color.g * color.a, color.a)

or straight / unpremultiplied

vec4 RGB = vec4 (color.r, color.g, color.b, color.a)

Note the difference in color channels.

I forget what unity skyboxes expect do behind the scenes because I don't do a lot of Unity dev (cc @brianchasalow ?) but the you might be able to fix this on the Syphon server client side by unpremultiplying , or pre-multiplying depending on the blending in your Syphon client.

**Comment by vade** [comment]
you can force unpremultiplied from a premultiplied by dividing each color component by the alpha btw. Typically you want to have a pipeline that chooses one approach. Syphon does not and never will adjust the alpha / color channels, its agnostic to premultiplied or unpremultiplied. This is good, but can cause confusion when you mix approaches like above.

**Comment by brianchasalow** [comment]
Can't you just stick a SyphonServerAlphaOne*.cs *image effect on the camera
you're trying to render the syphon server with? It'll write 1 to the alpha
channel- if the image looks correct in Unity but looks wrong in the app
that's receiving the Syphon texture, this is usually the correct answer,
unless you want to actually use the alpha channel in some meaningful way in
the app you're sharing the texture with, in which case you have to
carefully manage what is getting written to the alpha channel within Unity.
Brian

**Comment by VSLABKRS** [comment]
Yeah, sticking a SyphonServerAlphaOne*.cs *image effect on the camera helped! Thanks brianchasalow and vade!

**Comment by preamptive** [comment]
There is a great Unity implementation of Syphon here... 
https://github.com/keijiro/Funnel
fully compatible with the new OpenGL backend (OpenGL Core)
Unity 5.3 or later

**Comment by VSLABKRS** [comment]
Yes, it's nice, but only syphon-server, no client yet.
